### ISSUE TITLE
[Windows] FirstFrameReady happens while still showing a blank image (specific videos only)

### ISSUE THREAD

**Issue by TheloTheGreat** [RenderHeads/UnityPlugin-AVProVideo#679]
On Windows (but not Mac), when opening a particular new video, the FirstFrameReady event sometimes fires while we still see a blank image on the MediaPlayer's display (DisplayUGUI, for instance). The FirstFrameReady should only fire when a valid image from the new video really is on-screen.

To repro, on Windows AVPro 2.0.9 in Demo_MediaPlayer, in MediaPlayerUI.cs:

First download this video file and put it in "Assets/StreamingAssets/AVProVideoSamples/":
https://thelo.ca/doom_featured_anim.mp4

Add this to Start():

```
if (_mediaPlayer) _mediaPlayer.Events.AddListener(OnMediaPlayerEvent);
```

Add this new method:

```
public void OnMediaPlayerEvent(MediaPlayer mp, MediaPlayerEvent.EventType type, ErrorCode errorCode) {
	if (type == MediaPlayerEvent.EventType.FirstFrameReady) {
		Debug.LogError("First frame ready"); // Auto-pause the editor
	}
}
```

Add this to Update:

```
if (Input.GetKeyDown(KeyCode.Q)) {
	_mediaPlayer.OpenMedia(MediaPathType.RelativeToProjectFolder, "Assets/StreamingAssets/AVProVideoSamples/doom_featured_anim.mp4");
}
```

Then launch the scene, manually unpausing the editor once to allow the initial video to start playing a bit. Press the Q key to open the doom_featured_anim.mp4 video, and the editor will pause when the FirstFrameReady event happens.

Expected result: The editor pauses on the first frame of the video.
Actual result: Sometimes (maybe 40%?), the editor pauses on a blank frame.
This blank frame may then later update to show the first frame of the video, even if the editor is paused (!). That seems to happen more often if I wiggle the mouse over the video area. (Does this trigger a redraw of that part of the screen?)

Notes:
- This only seems to happen on Windows, not on Mac.
- This does NOT happen with all video files, only certain ones. It happens with doom_featured_anim.mp4, but does not happen with any of the video files in the AVPro demo. I don't know yet which ones cause this bad state.
- This does not happen 100% of the time, so you may need to resume editor playback, then press Q again a few times to see it really pause on a blank frame. I have a suspicion that constantly moving the mouse causes the issue to happen less frequently, but it's hard to really tell.
- The blank frame can update within a few seconds even if the editor is paused, which is pretty wild.
- As a crappy workaround that's probably a bad idea, I can prevent the issue by editing MediaPlayer_Events.cs:213 to require `GetTextureFrameCount() > 1` like this:

```
case MediaPlayerEvent.EventType.FirstFrameReady:
	// [MOZ 20/1/21] Removed HasMetaData check as preventing the event from being triggered on (i|mac|tv)OS
	//result = (_textureInterface != null && _controlInterface.CanPlay() /*&& _controlInterface.HasMetaData()*/ && _textureInterface.GetTextureFrameCount() > 0);
	result = (_textureInterface != null && _controlInterface.CanPlay() /*&& _controlInterface.HasMetaData()*/ && _textureInterface.GetTextureFrameCount() > 1); // THELO EDIT
	break;
```

**Comment by AndrewRH** [comment]
Hi,

Yes I think this is a bug that was introduced in 2.0.9 - the next version should fix it.

Thanks,

**Comment by TheloTheGreat** [comment]
Fix confirmed on 2.0.10 for Windows, thanks.
